Framework/DataObjects/src/FractionalRebinning.cpp

```diff
@@ -185,8 +185,8 @@ void calcTrapezoidYIntersections(
     ur = V2D(xAxis[x_end], mTop * xAxis[x_end] + cTop);
   }
 
-  const size_t nx = x_end - x_start;
-  const size_t ny = y_end - y_start;
+  const size_t nx = x_end - x_start + 1;
+  const size_t ny = y_end - y_start + 1;
   const double ll_x(ll.X()), ll_y(ll.Y()), ul_y(ul.Y());
   const double lr_x(lr.X()), lr_y(lr.Y()), ur_y(ur.Y());
   // Check if there is only a output single bin and inputQ is fully enclosed
```

Comment:
Is it correct that the next statement would now only fire if `xend==x_start` ans `y_end==y_start`?

Reply:
Doh! I think I completely missed the point. I just looked at the loop in the general code (`calcGeneralIntersection`) and thought that it meant that when `[xy]_start==[xy]_end` meant there is only one bin width, and that I had misunderstood what `[xy]_end` meant. But actually the code before was right. 

What `[xy]_start == [xy]_end` means is that the *bin width* in that direction is zero, and this was what was causing the segmentation fault.

So, actually we can fix the segfault without changing so much of the code. I'll push another commit to do this...